# HG changeset patch
# User Rik <rik@octave.org>
# Date 1382645833 25200
#      Thu Oct 24 13:17:13 2013 -0700
# Node ID e2f83152e64a2c1147a26da9590d7422d90b04a0
# Parent  94c7b2252490d48ae76d4423f967fdcd97f68326
print.m: Restore figure position first (in pixels) and then original units.

* scripts/plot/util/print.m: Restore figure position first (in pixels) and then
original units.

diff --git a/scripts/plot/util/print.m b/scripts/plot/util/print.m
--- a/scripts/plot/util/print.m
+++ b/scripts/plot/util/print.m
@@ -420,17 +420,23 @@ function print (varargin)
         opts = __gnuplot_print__ (opts);
       otherwise
         opts = __fltk_print__ (opts);
     endswitch
 
   unwind_protect_cleanup
     ## restore modified properties
     if (isstruct (props))
-      for n = 1:numel (props)
+      ## Restore figure position and units first
+      for n = 2:-1:1
+        if (ishandle (props(n).h))
+          set (props(n).h, props(n).name, props(n).value{1});
+        endif
+      endfor
+      for n = numel (props):-1:3
         if (ishandle (props(n).h))
           set (props(n).h, props(n).name, props(n).value{1});
         endif
       endfor
     endif
 
     ## Unlink temporary files
     for n = 1:numel (opts.unlink)
